***
## First we import pandas that we will use though out all 3 data sets

In [1]:
import pandas as pd

***
# First data set: Cars Data
This was the data set i used in my week 5 discussion board.
It is a dataset of about 400 cars with 8 characteristics such as horsepower, acceleration, etc.


Retrived from:  [Project Datasets Site](https://perso.telecom-paristech.fr/eagan/class/igr204/datasets)

***
We will import the data from the site into the `cars` variable

In [2]:
cars = pd.read_csv('https://perso.telecom-paristech.fr/eagan/class/igr204/data/cars.csv',sep=';',header=0,)
cars

Car     MPG Cylinders Displacement  \
0                               STRING  DOUBLE       INT       DOUBLE   
1            Chevrolet Chevelle Malibu    18.0         8        307.0   
2                    Buick Skylark 320    15.0         8        350.0   
3                   Plymouth Satellite    18.0         8        318.0   
4                        AMC Rebel SST    16.0         8        304.0   
5                          Ford Torino    17.0         8        302.0   
6                     Ford Galaxie 500    15.0         8        429.0   
7                     Chevrolet Impala    14.0         8        454.0   
8                    Plymouth Fury iii    14.0         8        440.0   
9                     Pontiac Catalina    14.0         8        455.0   
10                  AMC Ambassador DPL    15.0         8        390.0   
11                Citroen DS-21 Pallas       0         4        133.0   
12    Chevrolet Chevelle Concours (sw)       0         8        350.0   
13                    Ford Torino (sw)       0         8        351.0   
14             Plymouth Satellite (sw)       0         8        383.0   
15                  AMC Rebel SST (sw)       0         8        360.0   
16                 Dodge Challenger SE    15.0         8        383.0   
17                  Plymouth 'Cuda 340    14.0         8        340.0   
18               Ford Mustang Boss 302       0         8        302.0   
19               Chevrolet Monte Carlo    15.0         8        400.0   
20             Buick Estate Wagon (sw)    14.0         8        455.0   
21              Toyota Corolla Mark ii    24.0         4        113.0   
22                     Plymouth Duster    22.0         6        198.0   
23                          AMC Hornet    18.0         6        199.0   
24                       Ford Maverick    21.0         6        200.0   
25                        Datsun PL510    27.0         4        97.00   
26        Volkswagen 1131 Deluxe Sedan    26.0         4        97.00   
27                         Peugeot 504    25.0         4        110.0   
28                         Audi 100 LS    24.0         4        107.0   
29                            Saab 99e    25.0         4        104.0   
..                                 ...     ...       ...          ...   
377           Chevrolet Cavalier Wagon    27.0         4        112.0   
378          Chevrolet Cavalier 2-door    34.0         4        112.0   
379         Pontiac J2000 SE Hatchback    31.0         4        112.0   
380                     Dodge Aries SE    29.0         4        135.0   
381                    Pontiac Phoenix    27.0         4        151.0   
382               Ford Fairmont Futura    24.0         4        140.0   
383                     AMC Concord DL    23.0         4        151.0   
384                Volkswagen Rabbit l    36.0         4        105.0   
385                 Mazda GLC Custom l    37.0         4        91.00   
386                   Mazda GLC Custom    31.0         4        91.00   
387             Plymouth Horizon Miser    38.0         4        105.0   
388                     Mercury Lynx l    36.0         4        98.00   
389                   Nissan Stanza XE    36.0         4        120.0   
390                       Honda Accord    36.0         4        107.0   
391                     Toyota Corolla    34.0         4        108.0   
392                        Honda Civic    38.0         4        91.00   
393                 Honda Civic (auto)    32.0         4        91.00   
394                      Datsun 310 GX    38.0         4        91.00   
395              Buick Century Limited    25.0         6        181.0   
396  Oldsmobile Cutlass Ciera (diesel)    38.0         6        262.0   
397         Chrysler Lebaron Medallion    26.0         4        156.0   
398                     Ford Grenada l    22.0         6        232.0   
399                   Toyota Celica GT    32.0         4        144.0   
400                 

As we can see `row 0` is a guide, to inform us of how the data in the corresponding column should be represented. We do not need this row so we will remove it.

In [3]:
cars.drop(cars.index[0], inplace=True)
cars

Car   MPG Cylinders Displacement  \
1            Chevrolet Chevelle Malibu  18.0         8        307.0   
2                    Buick Skylark 320  15.0         8        350.0   
3                   Plymouth Satellite  18.0         8        318.0   
4                        AMC Rebel SST  16.0         8        304.0   
5                          Ford Torino  17.0         8        302.0   
6                     Ford Galaxie 500  15.0         8        429.0   
7                     Chevrolet Impala  14.0         8        454.0   
8                    Plymouth Fury iii  14.0         8        440.0   
9                     Pontiac Catalina  14.0         8        455.0   
10                  AMC Ambassador DPL  15.0         8        390.0   
11                Citroen DS-21 Pallas     0         4        133.0   
12    Chevrolet Chevelle Concours (sw)     0         8        350.0   
13                    Ford Torino (sw)     0         8        351.0   
14             Plymouth Satellite (sw)     0         8        383.0   
15                  AMC Rebel SST (sw)     0         8        360.0   
16                 Dodge Challenger SE  15.0         8        383.0   
17                  Plymouth 'Cuda 340  14.0         8        340.0   
18               Ford Mustang Boss 302     0         8        302.0   
19               Chevrolet Monte Carlo  15.0         8        400.0   
20             Buick Estate Wagon (sw)  14.0         8        455.0   
21              Toyota Corolla Mark ii  24.0         4        113.0   
22                     Plymouth Duster  22.0         6        198.0   
23                          AMC Hornet  18.0         6        199.0   
24                       Ford Maverick  21.0         6        200.0   
25                        Datsun PL510  27.0         4        97.00   
26        Volkswagen 1131 Deluxe Sedan  26.0         4        97.00   
27                         Peugeot 504  25.0         4        110.0   
28                         Audi 100 LS  24.0         4        107.0   
29                            Saab 99e  25.0         4        104.0   
30                            BMW 2002  26.0         4        121.0   
..                                 ...   ...       ...          ...   
377           Chevrolet Cavalier Wagon  27.0         4        112.0   
378          Chevrolet Cavalier 2-door  34.0         4        112.0   
379         Pontiac J2000 SE Hatchback  31.0         4        112.0   
380                     Dodge Aries SE  29.0         4        135.0   
381                    Pontiac Phoenix  27.0         4        151.0   
382               Ford Fairmont Futura  24.0         4        140.0   
383                     AMC Concord DL  23.0         4        151.0   
384                Volkswagen Rabbit l  36.0         4        105.0   
385                 Mazda GLC Custom l  37.0         4        91.00   
386                   Mazda GLC Custom  31.0         4        91.00   
387             Plymouth Horizon Miser  38.0         4        105.0   
388                     Mercury Lynx l  36.0         4        98.00   
389                   Nissan Stanza XE  36.0         4        120.0   
390                       Honda Accord  36.0         4        107.0   
391                     Toyota Corolla  34.0         4        108.0   
392                        Honda Civic  38.0         4        91.00   
393                 Honda Civic (auto)  32.0         4        91.00   
394                      Datsun 310 GX  38.0         4        91.00   
395              Buick Century Limited  25.0         6        181.0   
396  Oldsmobile Cutlass Ciera (diesel)  38.0         6        262.0   
397         Chrysler Lebaron Medallion  26.0         4        156.0   
398                     Ford Grenada l  22.0         6        232.0   
399                   Toyota Celica GT  32.0         4        144.0   
400                  Dodge Charger 2.2  36.0         4        135.0   
401                   Chevrolet Camaro  27.0         4     

***
Now we will look to see if we have and `Nan` values in our `cars` data set

In [4]:
cars.isnull().sum()

Car             0
MPG             0
Cylinders       0
Displacement    0
Horsepower      0
Weight          0
Acceleration    0
Model           0
Origin          0
dtype: int64

Luckily we find there are no `Nan` values.

***
# We will use this `cars` dataset to find out which `Origin` produces the most fule efficiant cars in the list, And out of those results, what most fule efficient car in that `Origin` has the most `Horsepower`.

***
Now in order to accomlish this, we will remove some of the columns in the `cars` data set, This will give us a cleaner data set that we will name `tidey_cars`. `tidey_cars` will only consit of the `car_id`, `Car`, `MPG`, `Horspower`, `Cylinders`, and `Origin`.

In [5]:
column_names = {'Car':'Car', 'MPG':'MPG', 'Horsepower':'Horsepower', 'Cylinders':'Cylinders','Origin':'Origin'}
tidey_cars = (cars.rename(columns=column_names)[['Car', 'MPG', 'Horsepower', 'Cylinders','Origin']].rename_axis(['car_id']).sort_index())
tidey_cars

Car   MPG Horsepower Cylinders  Origin
car_id                                                                      
1               Chevrolet Chevelle Malibu  18.0      130.0         8      US
2                       Buick Skylark 320  15.0      165.0         8      US
3                      Plymouth Satellite  18.0      150.0         8      US
4                           AMC Rebel SST  16.0      150.0         8      US
5                             Ford Torino  17.0      140.0         8      US
6                        Ford Galaxie 500  15.0      198.0         8      US
7                        Chevrolet Impala  14.0      220.0         8      US
8                       Plymouth Fury iii  14.0      215.0         8      US
9                        Pontiac Catalina  14.0      225.0         8      US
10                     AMC Ambassador DPL  15.0      190.0         8      US
11                   Citroen DS-21 Pallas     0      115.0         4  Europe
12       Chevrolet Chevelle Concours (sw)     0      165.0         8      US
13                       Ford Torino (sw)     0      153.0         8      US
14                Plymouth Satellite (sw)     0      175.0         8      US
15                     AMC Rebel SST (sw)     0      175.0         8      US
16                    Dodge Challenger SE  15.0      170.0         8      US
17                     Plymouth 'Cuda 340  14.0      160.0         8      US
18                  Ford Mustang Boss 302     0      140.0         8      US
19                  Chevrolet Monte Carlo  15.0      150.0         8      US
20                Buick Estate Wagon (sw)  14.0      225.0         8      US
21                 Toyota Corolla Mark ii  24.0      95.00         4   Japan
22                        Plymouth Duster  22.0      95.00         6      US
23                             AMC Hornet  18.0      97.00         6      US
24                          Ford Maverick  21.0      85.00         6      US
25                           Datsun PL510  27.0      88.00         4   Japan
26           Volkswagen 1131 Deluxe Sedan  26.0      46.00         4  Europe
27                            Peugeot 504  25.0      87.00         4  Europe
28                            Audi 100 LS  24.0      90.00         4  Europe
29                               Saab 99e  25.0      95.00         4  Europe
30                               BMW 2002  26.0      113.0         4  Europe
...                                   ...   ...        ...       ...     ...
377              Chevrolet Cavalier Wagon  27.0      88.00         4      US
378             Chevrolet Cavalier 2-door  34.0      88.00         4      US
379            Pontiac J2000 SE Hatchback  31.0      85.00         4      US
380                        Dodge Aries SE  29.0      84.00         4      US
381                       Pontiac Phoenix  27.0      90.00         4      US
382                  Ford Fairmont Futura  24.0      92.00         4      US
383                        AMC Concord DL  23.0          0         4      US
384                   Volkswagen Rabbit l  36.0      74.00         4  Europe
385                    Mazda GLC Custom l  37.0      68.00         4   Japan
386                      Mazda GLC Custom  31.0      68.00         4   Japan
387                Plymouth Horizon Miser  38.0      63.00         4      US
388                        Mercury Lynx l  36.0      70.00         4      US
389                      Nissan Stanza XE  36.0      88.00         4   Japan
390                          Honda Accord  36.0      75.00         4   Japan
391                        Toyota Corolla  34.0      70.00         4   Japan
392                           Honda Civic  38.0      67.00         4   Japan
393                    Honda Civic (auto)  32.0      67.00         4   Japan
394                         Datsun 310 GX  38.0      67.00         4   Japan
395                 Buick Century Limited  25.0      110.0         6      US
396     Oldsmobile Cutlass Ciera (

Now that we have a `tidey_cars` data set, to find the most fule efficant cars we will take the top 10 results and sort them by fule efficency. To do this we will sort them in descending order by `MPG`. We will name this list `most_fule_efficiant_cars`.

In [6]:
most_fule_efficiant_cars = tidey_cars.sort_values(by = 'MPG', ascending=False).head(10)
most_fule_efficiant_cars

Car   MPG Horsepower Cylinders  Origin
car_id                                                                    
35                             Hi 1200D   9.0      193.0         8      US
330                           Mazda GLC  46.6      65.00         4   Japan
337                 Honda Civic 1500 gl  44.6      67.00         4   Japan
333        Volkswagen Rabbit C (Diesel)  44.3      48.00         4  Europe
403                   Volkswagen Pickup  44.0      52.00         4  Europe
334          Volkswagen Dasher (diesel)  43.4      48.00         4  Europe
252     Volkswagen Rabbit Custom Diesel  43.1      48.00         4  Europe
317                   Volkswagen Rabbit  41.5      76.00         4  Europe
338                Renault Lecar Deluxe  40.9          0         4  Europe
332                          Datsun 210  40.8      65.00         4   Japan

We can see our first result is a V8 with 9.0 MPG which is definitly some bad data. However the other 9 results seem to be on track with an `MPG` in the 40+ range, and `Europe` seems to have a dominit hold.

To be sure we will do a quick count of the `Origins` in `most_fule_efficiant_cars`.

In [7]:
most_fule_efficiant_cars['Origin'].value_counts()

Europe    6
Japan     3
US        1
Name: Origin, dtype: int64

Yep, loos like `Europe` has 6 cars in the top 10 most fule efficient cars in the data set. Let clean this up further and return just those 6 cars in to the variable `europes_most_fule_efficiant_cars`

In [8]:
europes_most_fule_efficiant_cars = most_fule_efficiant_cars.loc[most_fule_efficiant_cars['Origin']=='Europe']
europes_most_fule_efficiant_cars

Car   MPG Horsepower Cylinders  Origin
car_id                                                                    
333        Volkswagen Rabbit C (Diesel)  44.3      48.00         4  Europe
403                   Volkswagen Pickup  44.0      52.00         4  Europe
334          Volkswagen Dasher (diesel)  43.4      48.00         4  Europe
252     Volkswagen Rabbit Custom Diesel  43.1      48.00         4  Europe
317                   Volkswagen Rabbit  41.5      76.00         4  Europe
338                Renault Lecar Deluxe  40.9          0         4  Europe

Now that we have this list of `europes_most_fule_efficiant_cars` lets find the one with the most horspower.

In [9]:
europes_most_fule_efficiant_cars.sort_values(by = 'Horsepower',ascending=False).head(1)

Car   MPG Horsepower Cylinders  Origin
car_id                                                      
317     Volkswagen Rabbit  41.5      76.00         4  Europe

Looks like `car_id`:317 is the winner. The `Volkswagen Rabbit` with and `MPG` of 41.5 has the most `Horsepower` with 76 horses.

## If you want the most fule efficient car, buy a European car, becuse they makes 6 of the top 10 most fule efficent cars in this list. And if you want one with the most Horspower the `Volkswagen Rabbit` is your pick with 76 Horses.

***


# Second Dataset: Ask A Manager Salary Survey 2019.

This was from Kathleen's Discussion board from Week 5.


It’s hard to get real-world information about what jobs pay, especially tailored to a particular industry or geographic region. In an effort to take some of the mystery out of salaries, ALISON GREEN ran a posts asking people to share how much money they make, their job, and their geographic region. Retrived from: [Ask A Manager Salary Survey 2019 (Responses) - Form Responses 1.csv](https://docs.google.com/spreadsheets/d/1rGCKXIKt-7l5gX06NAwO3pjqEHh-oPXtB8ihkp0vGWo/edit#gid=382484678)


We will import the data from the `Ask A Manager Salary Survey 2019 (Responses) - Form Responses 1.csv` file into the `salery_survey` variable.

In [10]:
salery_survey = pd.read_csv('Ask A Manager Salary Survey 2019 (Responses) - Form Responses 1.csv', header=1)
salery_survey

Timestamp How old are you?  \
0      4/24/2019 11:43:21            35-44   
1      4/24/2019 11:43:26            25-34   
2      4/24/2019 11:43:27            18-24   
3      4/24/2019 11:43:27            25-34   
4      4/24/2019 11:43:29            25-34   
5      4/24/2019 11:43:29            25-34   
6      4/24/2019 11:43:30            25-34   
7      4/24/2019 11:43:30            25-34   
8      4/24/2019 11:43:34            35-44   
9      4/24/2019 11:43:35            25-34   
10     4/24/2019 11:43:35            35-44   
11     4/24/2019 11:43:35            25-34   
12     4/24/2019 11:43:37            25-34   
13     4/24/2019 11:43:37            25-34   
14     4/24/2019 11:43:38            25-34   
15     4/24/2019 11:43:39            35-44   
16     4/24/2019 11:43:39            25-34   
17     4/24/2019 11:43:39            35-44   
18     4/24/2019 11:43:40            35-44   
19     4/24/2019 11:43:41            25-34   
20     4/24/2019 11:43:42            25-34   
21     4/24/2019 11:43:42            25-34   
22     4/24/2019 11:43:44            25-34   
23     4/24/2019 11:43:44            35-44   
24     4/24/2019 11:43:47            25-34   
25     4/24/2019 11:43:48            25-34   
26     4/24/2019 11:43:48            25-34   
27     4/24/2019 11:43:48            35-44   
28     4/24/2019 11:43:49            35-44   
29     4/24/2019 11:43:49            45-54   
...                   ...              ...   
33953   3/2/2020 16:51:39            35-44   
33954   3/2/2020 18:38:11            25-34   
33955   3/3/2020 10:02:44            18-24   
33956   3/3/2020 17:03:24            25-34   
33957   3/4/2020 13:12:03            35-44   
33958   3/4/2020 13:45:52            35-44   
33959   3/4/2020 13:53:22            25-34   
33960   3/4/2020 13:55:46            45-54   
33961   3/4/2020 13:57:19            35-44   
33962   3/4/2020 14:18:54            35-44   
33963   3/4/2020 14:35:56            25-34   
33964   3/4/2020 14:56:42            25-34   
33965   3/4/2020 15:05:44            18-24   
33966   3/4/2020 15:28:02            45-54   
33967   3/4/2020 16:59:23            35-44   
33968   3/4/2020 18:03:48            25-34   
33969   3/4/2020 18:12:33            35-44   
33970   3/4/2020 20:59:10            25-34   
33971   3/4/2020 21:41:16            25-34   
33972    3/5/2020 6:53:18            25-34   
33973   3/5/2020 13:17:59            25-34   
33974   3/5/2020 14:03:50            25-34   
33975   3/5/2020 14:08:21            25-34   
33976   3/5/2020 14:56:50            35-44   
33977   3/5/2020 15:17:01            25-34   
33978   3/5/2020 17:21:47            25-34   
33979   3/5/2020 17:46:08            25-34   
33980   3/5/2020 21:15:01            25-34   
33981   3/6/2020 14:25:44            25-34   
33982   3/7/2020 13:14:38            25-34   

                       What industry do you work in?  \
0                                         Government   
1                            Environmental nonprofit   
2                                    Market Research   
3                                      Biotechnology   
4                                         Healthcare   
5                             Information Management   
6                                         Nonprofit    
7                                   Higher Education   
8                                         Libraries    
9                                  Business Services   
10                       construction fastener sales   
11                                     Mental Health   
12                                Telecommunications   
13                                   Market research   
14                                         Education   
15                               Business Consulting   
16                                         Higher Ed   
17                                    communications   
18                                 Film & Television   
19                                         

Lets take deeper look at this file for `NaN`'s

In [11]:
salery_survey.isnull().sum()

Timestamp                                                                       0
How old are you?                                                                0
What industry do you work in?                                                 958
Job title                                                                       2
What is your annual salary?                                                     3
Please indicate the currency                                                    0
Where are you located? (City/state/country)                                  1742
How many years of post-college professional work experience do you have?        0
If your job title needs additional context, please clarify here:            26383
If "Other," please indicate the currency here:                              33710
dtype: int64

***
Looks like were going to have to clean up these column names, and fill in these `NaN` values. So lets do that and create the `tidey_salery_survey`.

In [12]:
column_names = {'How old are you?':'age',
                'What industry do you work in?':'industry',
                'Job title':'job',
                'What is your annual salary?':'salary',
                'Please indicate the currency':'currency',
                'Where are you located? (City/state/country)':'city/state/country',
                'How many years of post-college professional work experience do you have?':'years_exsperience'
               }
tidey_salery_survey = (salery_survey.rename(columns=column_names)
                       [['age','industry','job', 'salary','currency','city/state/country','years_exsperience']]
                       .rename_axis('entry#').sort_index())
tidey_salery_survey

age                                       industry  \
entry#                                                         
0       35-44                                     Government   
1       25-34                        Environmental nonprofit   
2       18-24                                Market Research   
3       25-34                                  Biotechnology   
4       25-34                                     Healthcare   
5       25-34                         Information Management   
6       25-34                                     Nonprofit    
7       25-34                               Higher Education   
8       35-44                                     Libraries    
9       25-34                              Business Services   
10      35-44                    construction fastener sales   
11      25-34                                  Mental Health   
12      25-34                             Telecommunications   
13      25-34                                Market research   
14      25-34                                      Education   
15      35-44                            Business Consulting   
16      25-34                                      Higher Ed   
17      35-44                                 communications   
18      35-44                              Film & Television   
19      25-34                                          media   
20      25-34                                     Publishing   
21      25-34                                     Non-profit   
22      25-34                               academic library   
23      35-44                               Higher education   
24      25-34                                         Events   
25      25-34                                            IT    
26      25-34    state government - business mgmt/facilities   
27      35-44                                 not for profit   
28      35-44                                       Pet care   
29      45-54  Federal contractor for science-related agency   
...       ...                                            ...   
33953   35-44                                           Tech   
33954   25-34                                           Tech   
33955   18-24                              Business Services   
33956   25-34                                            NaN   
33957   35-44                                      Libraries   
33958   35-44                               Public Libraries   
33959   25-34                                        Finance   
33960   45-54                               Public Libraries   
33961   35-44              Higher Education/Academic Library   
33962   35-44                                      Education   
33963   25-34                                        Library   
33964   25-34                         Biotech/Pharmaceutical   
33965   18-24                                      Libraries   
33966   45-54                                      Libraries   
33967   35-44                                 Public Library   
33968   25-34                                       Software   
33969   35-44                                 Public Library   
33970   25-34                               higher education   
33971   25-34                                     Non-Profit   
33972   25-34                                   ADVERTISING    
33973   25-34                              Digital marketing   
33974   25-34                                       Finance    
33975   25-34                                       Finance    
33976   35-44                               Library (Public)   
33977   25-34                                       Software   
33978   25-34                                          Legal   
33979   25-34                                            NaN   
33980   25-34                                       Library    
33981   25-34                                           Tech   
33982   25-34                             

Lets check our ` Nan`s and in the process we should make sure our column names changed.

In [13]:
tidey_salery_survey.isnull().sum()

age                      0
industry               958
job                      2
salary                   3
currency                 0
city/state/country    1742
years_exsperience        0
dtype: int64

Great that worked. Since we are dealling with a .csv file that we downloaded and dont want to accidently change values in the .csv file lets make a new variable for safty called `tidey_salery_survey_copy`. well work with this file to make changes from here on out.

In [14]:
tidey_salery_survey_copy = tidey_salery_survey.copy()
tidey_salery_survey_copy

age                                       industry  \
entry#                                                         
0       35-44                                     Government   
1       25-34                        Environmental nonprofit   
2       18-24                                Market Research   
3       25-34                                  Biotechnology   
4       25-34                                     Healthcare   
5       25-34                         Information Management   
6       25-34                                     Nonprofit    
7       25-34                               Higher Education   
8       35-44                                     Libraries    
9       25-34                              Business Services   
10      35-44                    construction fastener sales   
11      25-34                                  Mental Health   
12      25-34                             Telecommunications   
13      25-34                                Market research   
14      25-34                                      Education   
15      35-44                            Business Consulting   
16      25-34                                      Higher Ed   
17      35-44                                 communications   
18      35-44                              Film & Television   
19      25-34                                          media   
20      25-34                                     Publishing   
21      25-34                                     Non-profit   
22      25-34                               academic library   
23      35-44                               Higher education   
24      25-34                                         Events   
25      25-34                                            IT    
26      25-34    state government - business mgmt/facilities   
27      35-44                                 not for profit   
28      35-44                                       Pet care   
29      45-54  Federal contractor for science-related agency   
...       ...                                            ...   
33953   35-44                                           Tech   
33954   25-34                                           Tech   
33955   18-24                              Business Services   
33956   25-34                                            NaN   
33957   35-44                                      Libraries   
33958   35-44                               Public Libraries   
33959   25-34                                        Finance   
33960   45-54                               Public Libraries   
33961   35-44              Higher Education/Academic Library   
33962   35-44                                      Education   
33963   25-34                                        Library   
33964   25-34                         Biotech/Pharmaceutical   
33965   18-24                                      Libraries   
33966   45-54                                      Libraries   
33967   35-44                                 Public Library   
33968   25-34                                       Software   
33969   35-44                                 Public Library   
33970   25-34                               higher education   
33971   25-34                                     Non-Profit   
33972   25-34                                   ADVERTISING    
33973   25-34                              Digital marketing   
33974   25-34                                       Finance    
33975   25-34                                       Finance    
33976   35-44                               Library (Public)   
33977   25-34                                       Software   
33978   25-34                                          Legal   
33979   25-34                                            NaN   
33980   25-34                                       Library    
33981   25-34                                           Tech   
33982   25-34                             

Lets fill the `Nan` values with the `str` `'UNKNOWN'`.

In [15]:
tidey_salery_survey_copy.fillna(value='UNKNOWN', inplace=True)
tidey_salery_survey_copy

age                                       industry  \
entry#                                                         
0       35-44                                     Government   
1       25-34                        Environmental nonprofit   
2       18-24                                Market Research   
3       25-34                                  Biotechnology   
4       25-34                                     Healthcare   
5       25-34                         Information Management   
6       25-34                                     Nonprofit    
7       25-34                               Higher Education   
8       35-44                                     Libraries    
9       25-34                              Business Services   
10      35-44                    construction fastener sales   
11      25-34                                  Mental Health   
12      25-34                             Telecommunications   
13      25-34                                Market research   
14      25-34                                      Education   
15      35-44                            Business Consulting   
16      25-34                                      Higher Ed   
17      35-44                                 communications   
18      35-44                              Film & Television   
19      25-34                                          media   
20      25-34                                     Publishing   
21      25-34                                     Non-profit   
22      25-34                               academic library   
23      35-44                               Higher education   
24      25-34                                         Events   
25      25-34                                            IT    
26      25-34    state government - business mgmt/facilities   
27      35-44                                 not for profit   
28      35-44                                       Pet care   
29      45-54  Federal contractor for science-related agency   
...       ...                                            ...   
33953   35-44                                           Tech   
33954   25-34                                           Tech   
33955   18-24                              Business Services   
33956   25-34                                        UNKNOWN   
33957   35-44                                      Libraries   
33958   35-44                               Public Libraries   
33959   25-34                                        Finance   
33960   45-54                               Public Libraries   
33961   35-44              Higher Education/Academic Library   
33962   35-44                                      Education   
33963   25-34                                        Library   
33964   25-34                         Biotech/Pharmaceutical   
33965   18-24                                      Libraries   
33966   45-54                                      Libraries   
33967   35-44                                 Public Library   
33968   25-34                                       Software   
33969   35-44                                 Public Library   
33970   25-34                               higher education   
33971   25-34                                     Non-Profit   
33972   25-34                                   ADVERTISING    
33973   25-34                              Digital marketing   
33974   25-34                                       Finance    
33975   25-34                                       Finance    
33976   35-44                               Library (Public)   
33977   25-34                                       Software   
33978   25-34                                          Legal   
33979   25-34                                        UNKNOWN   
33980   25-34                                       Library    
33981   25-34                                           Tech   
33982   25-34                             

Lets confirm we no longer have any `NaN` Values.

In [16]:
tidey_salery_survey_copy.isnull().sum()

age                   0
industry              0
job                   0
salary                0
currency              0
city/state/country    0
years_exsperience     0
dtype: int64

Great we have no `Nan`s.

***
# Now we will try to use this data to find which age group in the US has the highest average salary.

Lets take a look at our age Groups.

In [17]:
tidey_salery_survey_copy['age'].value_counts()

25-34         15138
35-44         10268
45-54          4528
18-24          1861
55-64          1840
65 or over      297
under 18         51
Name: age, dtype: int64

Lets pull out all the data that is in USD curency, giving out US results.

In [18]:
us_salaries = tidey_salery_survey_copy.loc[tidey_salery_survey_copy['currency']== 'USD']
us_salaries

age                                       industry  \
entry#                                                         
0       35-44                                     Government   
1       25-34                        Environmental nonprofit   
2       18-24                                Market Research   
4       25-34                                     Healthcare   
5       25-34                         Information Management   
6       25-34                                     Nonprofit    
7       25-34                               Higher Education   
8       35-44                                     Libraries    
9       25-34                              Business Services   
10      35-44                    construction fastener sales   
11      25-34                                  Mental Health   
14      25-34                                      Education   
15      35-44                            Business Consulting   
16      25-34                                      Higher Ed   
19      25-34                                          media   
20      25-34                                     Publishing   
21      25-34                                     Non-profit   
22      25-34                               academic library   
23      35-44                               Higher education   
24      25-34                                         Events   
25      25-34                                            IT    
26      25-34    state government - business mgmt/facilities   
28      35-44                                       Pet care   
29      45-54  Federal contractor for science-related agency   
30      45-54                              County Government   
31      35-44                                     Non-profit   
32      25-34                                       Catering   
33      35-44                                   Construction   
34      45-54                            Property Management   
35      18-24                                     E-commerce   
...       ...                                            ...   
33946   18-24                                     Journalism   
33947   25-34                            Enterprise Software   
33949   45-54                                     Technology   
33951   25-34                           Computer Engineering   
33952   25-34                         Executive Compensation   
33953   35-44                                           Tech   
33954   25-34                                           Tech   
33955   18-24                              Business Services   
33956   25-34                                        UNKNOWN   
33957   35-44                                      Libraries   
33958   35-44                               Public Libraries   
33959   25-34                                        Finance   
33960   45-54                               Public Libraries   
33961   35-44              Higher Education/Academic Library   
33962   35-44                                      Education   
33963   25-34                                        Library   
33964   25-34                         Biotech/Pharmaceutical   
33965   18-24                                      Libraries   
33967   35-44                                 Public Library   
33968   25-34                                       Software   
33969   35-44                                 Public Library   
33970   25-34                               higher education   
33971   25-34                                     Non-Profit   
33972   25-34                                   ADVERTISING    
33973   25-34                              Digital marketing   
33976   35-44                               Library (Public)   
33977   25-34                                       Software   
33978   25-34                                          Legal   
33979   25-34                                        UNKNOWN   
33980   25-34                             

***
We will have to make sure this `salary` data is cleaned and converted to Integers so we can do some math on it later.

In [19]:
us_salaries.salary = pd.to_numeric(us_salaries['salary'], errors='coerce')

/Users/dennisawad/anaconda2/lib/python2.7/site-packages/pandas/core/generic.py:4405: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  self[name] = value


Lets take a look at what `us_salaries` look like now.

In [20]:
us_salaries

age                                       industry  \
entry#                                                         
0       35-44                                     Government   
1       25-34                        Environmental nonprofit   
2       18-24                                Market Research   
4       25-34                                     Healthcare   
5       25-34                         Information Management   
6       25-34                                     Nonprofit    
7       25-34                               Higher Education   
8       35-44                                     Libraries    
9       25-34                              Business Services   
10      35-44                    construction fastener sales   
11      25-34                                  Mental Health   
14      25-34                                      Education   
15      35-44                            Business Consulting   
16      25-34                                      Higher Ed   
19      25-34                                          media   
20      25-34                                     Publishing   
21      25-34                                     Non-profit   
22      25-34                               academic library   
23      35-44                               Higher education   
24      25-34                                         Events   
25      25-34                                            IT    
26      25-34    state government - business mgmt/facilities   
28      35-44                                       Pet care   
29      45-54  Federal contractor for science-related agency   
30      45-54                              County Government   
31      35-44                                     Non-profit   
32      25-34                                       Catering   
33      35-44                                   Construction   
34      45-54                            Property Management   
35      18-24                                     E-commerce   
...       ...                                            ...   
33946   18-24                                     Journalism   
33947   25-34                            Enterprise Software   
33949   45-54                                     Technology   
33951   25-34                           Computer Engineering   
33952   25-34                         Executive Compensation   
33953   35-44                                           Tech   
33954   25-34                                           Tech   
33955   18-24                              Business Services   
33956   25-34                                        UNKNOWN   
33957   35-44                                      Libraries   
33958   35-44                               Public Libraries   
33959   25-34                                        Finance   
33960   45-54                               Public Libraries   
33961   35-44              Higher Education/Academic Library   
33962   35-44                                      Education   
33963   25-34                                        Library   
33964   25-34                         Biotech/Pharmaceutical   
33965   18-24                                      Libraries   
33967   35-44                                 Public Library   
33968   25-34                                       Software   
33969   35-44                                 Public Library   
33970   25-34                               higher education   
33971   25-34                                     Non-Profit   
33972   25-34                                   ADVERTISING    
33973   25-34                              Digital marketing   
33976   35-44                               Library (Public)   
33977   25-34                                       Software   
33978   25-34                                          Legal   
33979   25-34                                        UNKNOWN   
33980   25-34                             

Well looks like in the conversion we had some bad values we couldnt conver that resulted in `NAN` values lets get rid of those well call this new dataset  `us_salaries_int`.

In [21]:
us_salaries_int = us_salaries.dropna()
us_salaries_int

age                                     industry  \
entry#                                                       
0       35-44                                   Government   
4       25-34                                   Healthcare   
8       35-44                                   Libraries    
9       25-34                            Business Services   
14      25-34                                    Education   
19      25-34                                        media   
26      25-34  state government - business mgmt/facilities   
32      25-34                                     Catering   
33      35-44                                 Construction   
35      18-24                                   E-commerce   
36      35-44                                        Media   
37      45-54                                 Design Build   
38      45-54                                     Academia   
49      25-34                                Mental health   
50      35-44                                   Networking   
51      25-34                   National Security/Academia   
58      35-44                               K-12 Education   
59      35-44                                    Wholesale   
62      45-54                                   Consulting   
65      25-34                                   Healthcare   
69      25-34                           Financial services   
71      35-44                                     Academia   
72      35-44                           Finance/Accounting   
74      35-44                                   Government   
77      25-34                                    Libraries   
82      35-44                                      Finance   
83      25-34                                   Accounting   
84      35-44                                   Healthcare   
85      25-34                                          Law   
88      25-34                             higher education   
...       ...                                          ...   
33916   25-34                                    Nonprofit   
33917   25-34                                     science    
33918   25-34                                  fundraising   
33919   25-34                                      Finance   
33921   25-34                         Software Engineering   
33923   25-34              Healthcare/Nonprofit/Government   
33925   45-54                                       Events   
33928   18-24                                     Software   
33930   25-34                                   Technology   
33932   45-54                                    Education   
33935   35-44                                    Archivist   
33937   18-24                                   Consulting   
33938   25-34                                Manufacturing   
33939   25-34                                         Tech   
33940   25-34                                    Marketing   
33942   25-34                                  Engineering   
33949   45-54                                   Technology   
33951   25-34                         Computer Engineering   
33953   35-44                                         Tech   
33954   25-34                                         Tech   
33955   18-24                            Business Services   
33956   25-34                                      UNKNOWN   
33958   35-44                             Public Libraries   
33959   25-34                                      Finance   
33962   35-44                                    Education   
33968   25-34                                     Software   
33970   25-34                             higher education   
33972   25-34                                 ADVERTISING    
33973   25-34                            Digital marketing   
33978   25-34                                        Legal   

                                                      job    salary currency  \
entry#                 

Lets sort this from Highest to lowes `salary`.

In [22]:
us_salaries_int.sort_values(by='salary', ascending = False)

age                               industry  \
entry#                                                      
24832        55-64                                   porn   
28465        45-54                                    Law   
28775        55-64                             Healthcare   
15792        45-54                    Financial Services    
22234        35-44                             Government   
15616        25-34                                     IT   
28746        45-54                                Finance   
29530        35-44                                Finance   
16607        35-44                             Healthcare   
13403        45-54                                Banking   
27684        35-44                                Finance   
31289        35-44                               Software   
14247        45-54                                 Retail   
14240        45-54                                 Retail   
28543        45-54                            Technology    
28678        45-54                       Basic materials    
16714     under 18                                   Porn   
25927        25-34                                UNKNOWN   
19480        55-64                             Consulting   
19483        55-64                             Consulting   
24295        35-44                             Technology   
15761        45-54                                     Ya   
18102        55-64                           Architecture   
20017        45-54                               medicine   
32943        35-44                                Medical   
16609        45-54                                    CPG   
16599        45-54                                    CPG   
16614        45-54                                    CPG   
28991        18-24                                UNKNOWN   
31474        35-44                                Finance   
...            ...                                    ...   
20776     under 18  How is this different from glassdoor?   
32259        25-34                                      A   
19315        35-44                              Logistics   
22903        35-44                                UNKNOWN   
27534        35-44                                UNKNOWN   
18235   65 or over                                Retired   
28757        35-44                             Child care   
16464        25-34                             Industrial   
24727        55-64                                   None   
20782        35-44                             Parenting    
18463        35-44                            Healthcare    
16135        45-54                                UNKNOWN   
16447        25-34                            Industrial    
17777        18-24                                Student   
24316     under 18                  stay at home industry   
27529        35-44                                UNKNOWN   
16869        45-54                                UNKNOWN   
17412        45-54                            Unemployed    
1357         35-44                                   none   
20725        25-34                             Care giver   
1739         25-34                               Student    
17423        45-54                            Unemployed    
16456        25-34                          Construction    
15287        35-44                         Homeschool mom   
15674        35-44                             Unemployed   
18757        25-34                                  Legal   
28434        55-64                             Unemployed   
25546        55-64                                UNKNOWN   
23173        45-54                                   None   
12313        45-54                                     IT   

                             job      salary currency  \
entry#                                                  
24832               chief jizzer  10000000.0      USD   


Were gona have to use `numpy` to do some math, so lets import that to help us in creating our `pivot_table`

In [23]:
import numpy as np

Lets creat a pivot table  with the`mean` saleries for each age group.

In [24]:
table= pd.pivot_table(us_salaries_int, values='salary', index='age', aggfunc={'salary':np.mean})
#we format flotes to make the salaries more readable
pd.options.display.float_format = '{:.2f}'.format
table

salary
age                 
18-24       63359.35
25-34       88972.70
35-44      119399.65
45-54      138442.83
55-64      148665.80
65 or over 114610.52
under 18    94253.00

## Looks like the `55-64` age group has the highest average salary of `$148,665.80`

***


Lets do another pivot table to see what numer of years exsperience makes the most money in the survey.

In [25]:
table2= pd.pivot_table(us_salaries_int, values='salary', index='years_exsperience', aggfunc={'salary':np.mean})
table2

salary
years_exsperience          
1 year or less     75144.61
11 - 20 years     122605.08
2 - 4 years        72093.49
21 - 30 years     150871.96
31 - 40 years     149892.98
41 years or more  126408.52
5-7 years          88409.51
8 - 10 years      101179.99

### Looks like the highest average salery belongs to those with 21-30 years of exsperience.
***

# Third Dataset: Film data


This was the data set used in Taha's  week 5 discussion board. This data has about 1600 movies with properties such as length, main actor and actress, director and popularity.

Retrived from:  [Project Datasets Site](https://perso.telecom-paristech.fr/eagan/class/igr204/datasets)

## We will use this data to list the `Drama` movies that came out in the `1980'S`.

First we will import this dataset from the website in to the variable `films`.

In [26]:
films = pd.read_csv('https://perso.telecom-paristech.fr/eagan/class/igr204/data/film.csv', sep=';')
films

Year Length                                     Title          Subject  \
0      INT    INT                                    STRING              CAT   
1     1990    111                   Tie Me Up! Tie Me Down!           Comedy   
2     1991    113                                High Heels           Comedy   
3     1983    104                            Dead Zone, The           Horror   
4     1979    122                                      Cuba           Action   
5     1978     94                            Days of Heaven            Drama   
6     1983    140                                 Octopussy           Action   
7     1984    101                              Target Eagle           Action   
8     1989     99    American Angels: Baptism of Blood, The            Drama   
9     1985    104                                    Subway            Drama   
10    1990    149                           Camille Claudel            Drama   
11    1982    188                       Fanny and Alexander            Drama   
12    1982    117               Tragedy of a Ridiculous Man            Drama   
13    1966    103                           A Man & a Woman            Drama   
14    1986    112       A Man & a Woman: Twenty Years Later            Drama   
15    1966    103                     Un Hombre y una Mujer            Drama   
16    1985    112                       Official Story, The            Drama   
17    1976    150            Lindbergh Kidnapping Case, The            Drama   
18    1929     84                                 Blackmail          Mystery   
19    1963    109                            Donovan's Reef           Comedy   
20    1988    110               Tucker: The Man & His Dream            Drama   
21    1988    101                                  Scrooged           Comedy   
22    1981    116                   Raiders of the Lost Ark           Action   
23    1987    101                          Running Man, The  Science Fiction   
24    1991    105                                Predator 2           Action   
25    1988    127                                    Colors            Drama   
26    1990     97                                  Zandalee            Drama   
27    1988    108                           Miles from Home            Drama   
28    1980    NaN                      Happy Birthday to Me           Horror   
29    1989     88                              Final Notice          Mystery   
...    ...    ...                                       ...              ...   
1630  1934     54                                 Hell Town          Western   
1631  1932    NaN                         Hurricane Express          Western   
1632  1932    210                    Hurricane Express, The           Action   
1633  1965    165                             In Harm's Way              War   
1634  1991    NaN     John Wayne Collection, Red River, The              War   
1635  1992    NaN    John Wayne Collector's Limited Edition              War   
1636  1991    NaN                      John Wayne Four Pack          Western   
1637  1939    112  John Wayne Matinee Double Feature, No. 2          Western   
1638  1939    110  John Wayne Matinee Double Feature, No. 3          Western   
1639  1938    110  John Wayne Matinee Double Feature, No. 4          Western   
1640  1990    NaN                       John Wayne Six Pack          Western   
1641  1991    NaN      John Wayne Western Greats, Rio Bravo          Western   
1642  1991     56                         King of the Pecos          Western   
1643  1992     59                          Lawless Frontier          Western   
1644  1991     52                     Lawless Frontier, The          Western   
1645  1991     56                     Lawless Nineties, The          Western   
1646  1934     54                               Lucky Texan          Western   
1647  1992    112                                       McQ           Action   
1648  

As we can see row 0 is a guide, to inform us of how the data in the corresponding column should be represented. We do not need this row so we will remove it just like we did in out first dataset.

In [27]:
films.drop(films.index[0], inplace=True)
films

Year Length                                     Title          Subject  \
1     1990    111                   Tie Me Up! Tie Me Down!           Comedy   
2     1991    113                                High Heels           Comedy   
3     1983    104                            Dead Zone, The           Horror   
4     1979    122                                      Cuba           Action   
5     1978     94                            Days of Heaven            Drama   
6     1983    140                                 Octopussy           Action   
7     1984    101                              Target Eagle           Action   
8     1989     99    American Angels: Baptism of Blood, The            Drama   
9     1985    104                                    Subway            Drama   
10    1990    149                           Camille Claudel            Drama   
11    1982    188                       Fanny and Alexander            Drama   
12    1982    117               Tragedy of a Ridiculous Man            Drama   
13    1966    103                           A Man & a Woman            Drama   
14    1986    112       A Man & a Woman: Twenty Years Later            Drama   
15    1966    103                     Un Hombre y una Mujer            Drama   
16    1985    112                       Official Story, The            Drama   
17    1976    150            Lindbergh Kidnapping Case, The            Drama   
18    1929     84                                 Blackmail          Mystery   
19    1963    109                            Donovan's Reef           Comedy   
20    1988    110               Tucker: The Man & His Dream            Drama   
21    1988    101                                  Scrooged           Comedy   
22    1981    116                   Raiders of the Lost Ark           Action   
23    1987    101                          Running Man, The  Science Fiction   
24    1991    105                                Predator 2           Action   
25    1988    127                                    Colors            Drama   
26    1990     97                                  Zandalee            Drama   
27    1988    108                           Miles from Home            Drama   
28    1980    NaN                      Happy Birthday to Me           Horror   
29    1989     88                              Final Notice          Mystery   
30    1979    110                                   Quintet            Drama   
...    ...    ...                                       ...              ...   
1630  1934     54                                 Hell Town          Western   
1631  1932    NaN                         Hurricane Express          Western   
1632  1932    210                    Hurricane Express, The           Action   
1633  1965    165                             In Harm's Way              War   
1634  1991    NaN     John Wayne Collection, Red River, The              War   
1635  1992    NaN    John Wayne Collector's Limited Edition              War   
1636  1991    NaN                      John Wayne Four Pack          Western   
1637  1939    112  John Wayne Matinee Double Feature, No. 2          Western   
1638  1939    110  John Wayne Matinee Double Feature, No. 3          Western   
1639  1938    110  John Wayne Matinee Double Feature, No. 4          Western   
1640  1990    NaN                       John Wayne Six Pack          Western   
1641  1991    NaN      John Wayne Western Greats, Rio Bravo          Western   
1642  1991     56                         King of the Pecos          Western   
1643  1992     59                          Lawless Frontier          Western   
1644  1991     52                     Lawless Frontier, The          Western   
1645  1991     56                     Lawless Nineties, The          Western   
1646  1934     54                               Lucky Texan          Western   
1647  1992    112                                       McQ           Action   
1648  

Lets take a look at our counts for type of movies using the `Subject` column

In [28]:
films.Subject.value_counts()

Drama              655
Comedy             385
Action             205
Western            119
Mystery            107
Horror              57
Music               43
Science Fiction     38
War                 34
Westerns             6
Adventure            4
Fantasy              1
Short                1
Romance              1
Crime                1
Name: Subject, dtype: int64

Now lets make sure all the rows that should be integers, are integers. So we will convert `Year`, `Length`, and `Popularity` to integers.

In [29]:
films.Year = pd.to_numeric(films['Year'], errors='coerce')
films.Length = pd.to_numeric(films['Length'], errors='coerce')
films.Popularity = pd.to_numeric(films['Popularity'], errors='coerce')


Now we can query our `films` dataset for `Drama` films from the 1980's (so all years between `1980` and `1989`) and we will sort them according to `popularity`.

In [30]:
films.loc[(films['Year'] >= 1980) & (films['Year']<= 1989) & (films['Subject']=='Drama')].sort_values(by='Popularity', ascending = False)

Year  Length                               Title Subject  \
1107  1985  161.00                       Out of Africa   Drama   
430   1988     nan                        Five Corners   Drama   
1023  1982  128.00             Ballad of Narayama, The   Drama   
332   1986  120.00                    Jean de Florette   Drama   
333   1989  127.00                Fat Man & Little Boy   Drama   
172   1982  108.00                      King of Comedy   Drama   
1567  1982  208.00                             Othello   Drama   
1294  1981  118.00                                Taps   Drama   
398   1984  112.00                 Places in the Heart   Drama   
47    1986  107.00                        Duet for One   Drama   
58    1981  106.00                Woman Next Door, The   Drama   
512   1984  100.00                         Swing Shift   Drama   
11    1982  188.00                 Fanny and Alexander   Drama   
794   1987  130.00                             Matewan   Drama   
1504  1982  150.00                 A Question of Honor   Drama   
602   1981  132.00                            Mephisto   Drama   
993   1980   92.00                          Rodeo Girl   Drama   
71    1987  102.00                     Babette's Feast   Drama   
663   1984  106.00             A Year of the Quiet Sun   Drama   
462   1986  122.00                            Salvador   Drama   
269   1988  119.00                  Dangerous Liaisons   Drama   
257   1989   90.00                    Never on Tuesday   Drama   
775   1986  130.00               El Guerrero Solitario   Drama   
428   1988   98.00                       Stealing Home   Drama   
703   1989  116.00                          In Country   Drama   
401   1981  116.00                   Absence of Malice   Drama   
626   1981  195.00                                Reds   Drama   
797   1988  109.00                      Unsettled Land   Drama   
1320  1989   90.00                   American Autobahn   Drama   
977   1986   96.00                      Pretty in Pink   Drama   
...    ...     ...                                 ...     ...   
115   1986  117.00                 Nine & a Half Weeks   Drama   
1124  1989   99.00                  Driving Miss Daisy   Drama   
959   1989  122.00                           Quo Vadis   Drama   
782   1986  119.00                 Color of Money, The   Drama   
9     1985  104.00                              Subway   Drama   
87    1987  100.00              Sammy & Rosie Get Laid   Drama   
974   1987  153.00                   Empire of the Sun   Drama   
1371  1984  141.00                 Killing Fields, The   Drama   
497   1987  126.00                         Wall Street   Drama   
1362  1981  124.00                    Chariots of Fire   Drama   
174   1984  158.00                             Amadeus   Drama   
587   1989  121.00               Enemies, a Love Story   Drama   
319   1989  120.00                  Do the Right Thing   Drama   
505   1987   96.00               Nights in White Satin   Drama   
1519  1982  136.00                              Danton   Drama   
176   1988  172.00  Unbearable Lightness of Being, The   Drama   
770   1980   94.00                     Out of the Blue   Drama   
684   1983   94.00                         Rumble Fish   Drama   
1177  1984  101.00                   Crimes of Passion   Drama   
84    1980  124.00                   Elephant Man, The   Drama   
1068  1987   99.00                        River's Edge   Drama   
466   1983  109.00                   Star Chamber, The   Drama   
559   1986  114.00                            Hoosiers   Drama   
1539  1981  127.00            Taming of the Shrew, The   Drama   
679   1988  116.00                    Running on Empty   Drama   
708   1986  105.00                              Ratboy   Drama   
1246  1982  125.00            An Officer & a Gentleman   Drama   
1296  1987  164.00                   Last Emperor, The   Drama   
892   1983   72.00                 After the R

## SO it looks like the top `Drama` film from the 1980's is actually a three way tie between `Outh of Africa` in 1985, `Five Corners` in 1988, and `The Ballad of Narayama` in 1982.